lib/grpc/client/resolver/dns.ex

```diff
@@ -0,0 +1,55 @@
+defmodule GRPC.Client.Resolver.DNS do
+  @moduledoc """
+  DNS Resolver for gRPC targets, supporting dynamic updates via a GenServer.
+
+  Resolves `dns://host[:port]` targets. Fetches A/AAAA records and optional
+  `_grpc_config.<host>` TXT records for ServiceConfig.
+
+  This implementation maintains an internal cache of addresses and service config,
+  and refreshes them periodically.
+  """
+  @behaviour GRPC.Client.Resolver
+
+  alias GRPC.Client.ServiceConfig
+
+  @impl GRPC.Client.Resolver
+  def resolve(target) do
+    uri = URI.parse(target)
+    host = uri.host || target
+    port = uri.port || 50051
```

Comment:
Why 50051? Should this be configurable?

Reply:
This is the commonly used port for gRpcs but of course, this should be configurable in the final version.